middle_end/flambda2/simplify/simplify_let_cont_expr.ml

```diff
@@ -898,11 +939,30 @@ let simplify_recursive_let_cont_handlers ~simplify_expr ~denv_before_body
       (after_downwards_traversal_of_one_recursive_let_cont_handler cont
          unboxing_decisions params ~original_cont_scope ~down_to_up)
 
+let rebuild_recursive_let_cont_expr art ~body ~free_names_of_body ~handlers =
+  let free_conts_of_body =
+    Name_occurrences.continuations_including_in_trap_actions free_names_of_body
+  in
+  let is_cont_used cont = Continuation.Set.mem cont free_conts_of_body in
+  match handlers with
+  | Non_recursive_handler (cont, handler) ->
+    if is_cont_used cont
+    then
+      RE.create_non_recursive_let_cont art cont handler ~body
+        ~free_names_of_body
+    else body
+  | Recursive_handlers rec_handlers ->
+    if Continuation.Map.exists (fun cont _ -> is_cont_used cont) rec_handlers
+    then RE.create_recursive_let_cont art rec_handlers ~body
+    else body
```

Comment:
It'd be better to do this decision on the way down, to avoid the problem of having a `uacc` which has recorded information on a piece of code that we delete (which has create bug previously, e.g. unbound symbol at linking time).